benchmarks/imagenet/resnet50/swav.py

```diff
@@ -0,0 +1,201 @@
+import math
+from typing import List, Tuple
+
+import torch
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import Identity, ModuleList
+from torch.nn import functional as F
+from torchvision.models import resnet50
+
+from lightly.loss.memory_bank import MemoryBankModule
+from lightly.loss.swav_loss import SwaVLoss
+from lightly.models.modules import SwaVProjectionHead, SwaVPrototypes
+from lightly.models.utils import get_weight_decay_parameters
+from lightly.transforms import SwaVTransform
+from lightly.utils.benchmarking import OnlineLinearClassifier
+from lightly.utils.lars import LARS
+from lightly.utils.scheduler import CosineWarmupScheduler
+
+CROP_COUNTS: Tuple[int, int] = (2, 6)
+
+
+class SwAV(LightningModule):
+    def __init__(self, batch_size_per_device: int, num_classes: int) -> None:
+        super().__init__()
+        self.save_hyperparameters()
+        self.batch_size_per_device = batch_size_per_device
+
+        resnet = resnet50()
+        resnet.fc = Identity()  # Ignore classification head
+        self.backbone = resnet
+        self.projection_head = SwaVProjectionHead()
+        self.prototypes = SwaVPrototypes(n_steps_frozen_prototypes=1)
+        self.criterion = SwaVLoss(sinkhorn_gather_distributed=True)
+        self.online_classifier = OnlineLinearClassifier(num_classes=num_classes)
+
+        # Use a queue for small batch sizes (<= 256).
+        self.start_queue_at_epoch = 15
+        self.n_batches_in_queue = 15
+        self.queues = ModuleList(
+            [
+                MemoryBankModule(
+                    size=self.n_batches_in_queue * self.batch_size_per_device
+                )
+                for _ in range(CROP_COUNTS[0])
+            ]
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        return self.backbone(x)
+
+    def project(self, x: Tensor) -> Tensor:
+        x = self.projection_head(x)
+        return F.normalize(x, dim=1, p=2)
+
+    def training_step(
+        self, batch: Tuple[List[Tensor], Tensor, List[str]], batch_idx: int
+    ) -> Tensor:
+        # Normalize the prototypes so they are on the unit sphere.
+        self.prototypes.normalize()
+
+        # The multi-crop dataloader returns a list of image crops where the
+        # first few items are high resolution crops and the rest are low
+        # resolution crops.
+        multi_crops, targets, _ = batch
```

Comment:
```suggestion
        multi_crops, targets = batch[0], batch[1]
```

This way it is also compatible with torch datasets.